gpiozero/internal_devices.py

```diff
@@ -481,22 +480,23 @@ def usage(self):
         """
         Returns the current disk usage in percentage.
         """
-        # XXX Use os.statvfs?
-        df = subprocess.Popen(['df', '--output=pcent', self.filesystem], stdout=subprocess.PIPE)
-        try:
-            output = df.communicate()[0].split(b'\n')[1].split()[0]
-            return float(output.decode('UTF-8').rstrip('%'))
-        finally:
-            df.stdout.close()
-            df.wait()
+        return self.value * 100
 
     @property
     def value(self):
         """
         Returns the current disk usage as a value between 0.0 and 1.0 by
         dividing :attr:`usage` by 100.
         """
-        return self.usage / 100.0
+        # This slightly convoluted calculation is equivalent to df's "Use%";
+        # it calculates the percentage of FS usage as a proportion of the
+        # space available to *non-root users*. Technically this means it can
+        # exceed 100% (when FS is filled to the point that only root can write
+        # to it), hence the clamp.
+        vfs = os.statvfs(self.filesystem)
+        used = vfs.f_blocks - vfs.f_bfree
+        total = used + vfs.f_bavail
+        return min(1.0, used / total)
```

Comment:
Could `total` ever be 0 ?

Reply:
Good question - I took a brief wade thru df and coreutils' code when writing this (to make sure my method of matching df's Use% output was equivalent) but I didn't notice any particular handling for div-by-0 (mind you I wasn't specifically looking so I may've missed it). I did note (from comments in that code) that under weird circumstances they can be negative but frankly I filed that under "weird enough that I don't care until someone complains"!

Reply:
I found this the other day, which also seems to corroborate things https://stackoverflow.com/questions/4965355/converting-statvfs-to-percentage-free-correctly